### ISSUE TITLE
Latest prerelease shows WW II Global 1940 Balanced Mod3 x2, release doesn't

### ISSUE THREAD

**Issue by tvleavitt** [triplea-game/triplea#7718]
### How can the problem be recreated?

Click "Select Map", scroll down to where the WW II mods will show. 

Release:

![2020-09-19 20_56_33-TripleA-balancedmodx2-releaseversionnotvisible](https://user-images.githubusercontent.com/4445057/93693948-431ca300-fabb-11ea-8ac0-bc1b407c680f.png)

Prerelease:

![2020-09-19 20_56_33-TripleA-balancedmodx2](https://user-images.githubusercontent.com/4445057/93693950-46179380-fabb-11ea-8f92-2543b002bc46.png)

### Do you have any ideas for an expected fix?

Behavior is consistent.

### Attach a Save Game

N/A

### If playing a prerelease, which version are you using?

Game Version: 2.3.21674

### If playing a prerelease, does this happen on the latest release?

- [] yes
- [x] no

**Comment by Cernelius** [comment]
Are they both games of the same map?

Isn't this just another case of https://github.com/triplea-game/triplea/issues/7089?

**Comment by panther2** [comment]
FWIW I cannot reproduce the issue:

![gl](https://user-images.githubusercontent.com/20812775/93716638-2e9ae200-fb71-11ea-8809-20cca579b73d.PNG)

**Comment by tvleavitt** [comment]
> Are they both games of the same map?
> 
> Isn't this just another case of #7089?

@Cernelius You're probably right. However, my point is that the same set of maps produces different behavior. Something has changed in the way they are parsed or the list is generated that exposed this issue more directly. Maybe the old mechanism was silently de-duping the list, and the new one isn't?

@panther2 At a guess, maybe you haven't downloaded the same set of maps that I have, somehow? I don't know which map package downloads include identically named maps.

**Comment by panther2** [comment]
@tvleavitt 

> At a guess, maybe you haven't downloaded the same set of maps that I have, somehow? I don't know which map package downloads include identically named maps.

That might be. I haven't installed anything that includes another xml with a line
`<info name="World War II Global 1940 Balanced Mod3" version="4.0"/>`

**Comment by DanVanAtta** [comment]
@tvleavitt the very latest will info log the name of each map loaded and location. Would you mind updating, launching the game, going into "settings > game", setting "show console to be true", and then relaunching. Then please let us know if the path to the two games with the same name is the same or not. If the paths are not the same, then the new behavior is technically an improvement as you do have distinct games but with different names.

**Comment by tvleavitt** [comment]
The path is: click on "Engine Preferences" (which launches a window entitled "Settings", TripleA's usual wildly inconsistent UI, ugh) and then clicking on the "Game" tab, setting "Show Console" to "True", and hitting "Save". The window, "Triple A Console" shows up in the background, buried under other windows, and even as a technical person, it is totally non-obvious what I'm supposed to be doing with this window. I click on various buttons, set "Normal Logging" and "Debug Logging", and I see nothing that looks like a list of game file paths.

I'm thinking that before I'm able to show console and view logs, the map pre-load process has completed, and nothing I see in the console will be useful? Because I've searched for everything logical and it nothing shows up in the console window when running the latest prerelease.

However, I did look in the downloadedMaps folder, and it looks like there's only one zip file with a matching name, when I looked, I saw this:

![2020-09-20 18_16_15-TripleAwindowsexplorerfilelistingview](https://user-images.githubusercontent.com/4445057/93726959-acf58500-fb6d-11ea-9a99-079c19a9783b.png)

Contents of ww2global40_balanced2_0.xml

<info version="4.0" name="World War II Global 1940 Balanced Mod3"/>
<loader javaClass="games.strategy.triplea.TripleA"/>
<triplea minimumVersion="1.8"/>
<diceSides value="6"/>

and contents of ww2global40_balanced2_0X.xml

<info version="3.4" name="World War II Global 1940 Balanced Mod3"/>
<loader javaClass="games.strategy.triplea.TripleA"/>
<triplea minimumVersion="1.8"/>
<diceSides value="6"/>

Perhaps this is the source of the problem?

**Comment by panther2** [comment]
@tvleavitt It looks like you have an outdated map-zip.
It should only contain these xml-files:

![xml](https://user-images.githubusercontent.com/20812775/93752751-63ee1100-fbff-11ea-857b-019a0619c3da.PNG)

**Comment by tvleavitt** [comment]
My "Date Modified" shows June 22, 2020; am I reading yours as July 11th, 2020? It's not prompting me to update when I hit "Download Maps". I could just remove and download again, but maybe that's the real bug: updates not being detected properly?

**Comment by panther2** [comment]
Most likely the YAML-Update had been missed, when that map had been updated.

**Comment by DanVanAtta** [comment]
Re: console, @tvleavitt if you set it to open, it'll launch on game start. Closing the console window will turn off the console and it won't launch on startup.

**Comment by tvleavitt** [comment]
O.K. So, this a process issue with the map maker's update, not a "bug" per se, then?

Maybe a "force check for updates" option might be useful since most end users aren't going to be aware they're using an outdated map under these circumstances?

**Comment by tvleavitt** [comment]
Deleting and reinstalling fixed the problem.

Buglet: the new one time loading of maps on start means that a deleted map still shows in "Select Map" and produces an error when selected; I wound up with three listed versions of Balanced Mod 3 until I shut down and restarted. Is there a way to trigger updating the list on delete of a map?

**Comment by panther2** [comment]
@tvleavitt 

> O.K. So, this a process issue with the map maker's update, not a "bug" per se, then?
>
> Maybe a "force check for updates" option might be useful since most end users aren't going to be aware they're using an outdated map under these circumstances?

Indeed that is one of the weeknesses of map-(update-) management by the triplea_maps.yaml. I remember we have discussed this from time to time. Unfortunately I cannot remember the results...

**Comment by DanVanAtta** [comment]
@panther2 The XML download list pre 1.9 had an equivalent version number field, I think that has been a problem since project inception.

There have been two debates on the topic, whether the version number in YAML should match the one in XML (that was previous practice, it is convenient but deceptive to think they are actually related). The second debate is whether an exact version should be downloaded or always the latest. An exact version is preferred but adds complexity. Given that map makers were already having difficulties with the transition to a self-service system where they could launch their own maps, and the lack of general usage because of that as well, a simpler approach was taken.

**Comment by DanVanAtta** [comment]
@tvleavitt yeah, now that map lists are cached, changes made directly to the file system are not represented. I was thinking a 'refresh' button might be nice to have. For now, I think it's okay, if someone modifies the file system directly, then I think they perhaps can expect problems and would not be too troubled to restart.

**Comment by tvleavitt** [comment]
I didn't actually modify the filesystem directly, I used the "Download Maps" function to remove and then install the same map. Admittedly, an unusual scenario. The new install appeared, but the removed map elements were still in the list until reboot, so you're picking up one set of changes, but not the other.

**Comment by DanVanAtta** [comment]
> Maybe a "force check for updates" option might be useful since most end users aren't going to be aware they're using an outdated map under these circumstances?

That perhaps would not help.This is a surprisingly difficult problem to solve:
1. Cultural/political: maps were classically handled by admins completely. On transition to github, it was more popular to conflate and complain about git and github than it was to understand the transition, the benefits and reasoning.
2. Documentation does not have a consolidated home, is scattered, duplicated, and is not readily updated.
3. Continued lack of understanding, which stems from (2)
4. Complicated process, this is a historical problem and in part is why admins just dealt with it (even though it reduced efficiency and there was long backlog of map requests that formed).

The 'stuka' project for maps to be moved to a server and database would help.

**Comment by DanVanAtta** [comment]
@tvleavitt remove maps not updating cache is a defect.

**Comment by DanVanAtta** [comment]
Does this still happen on the latest? The select map window shows the path of the XML file now, that could help debug if it is still occurring.

**Comment by DanVanAtta** [comment]
Map system is largely re-written in 2.7, mooting this issue
